ext/datadog_cov/datadog_cov.c

```diff
@@ -159,6 +120,12 @@ static VALUE dd_cov_start(VALUE self)
   return self;
 }
 
+static int insert_into_result(st_data_t filename, st_data_t _val, st_data_t coverage)
+{
+  rb_hash_aset((VALUE)coverage, rb_str_new2((char *)filename), Qtrue);
+  return ST_CONTINUE;
+}
```

Comment:
Thanks for pointing it out! yes, I kind of suspected that this is a bit fishy.

I will try the approach with replacing `st.h` with Ruby hash again but storing VALUE returned by `rb_profile_frames` to avoid creating new strings: it could provide me the same performance win.

Reply:
:+1: If you're going in that direction, remember to use the limit arguments in `rb_profile_frames` to get only the top entry.